test/wh_test_crypto.c

```diff
@@ -116,7 +116,83 @@ int whTest_CryptoClientConfig(whClientConfig* config)
         WH_ERROR_PRINT("KEY CACHE/EXPORT FAILED TO MATCH\n");
         goto exit;
     }
-    /* test evict */
+    /* test cache with duplicate keyId for a different user */
+    WH_TEST_RETURN_ON_FAIL(wh_Client_CommClose(client));
+    client->comm->client_id = 2;
```

Comment:
This shouldn't work this way.  The server should be applying the client id that it stored after the client invoked wh_Client_CommInit(c);   I understand this is for testing only, but I don't see how comm->client_id is being used at all in the code.  Are we sure this is working the way you expected?